### PR TITLE
Pass http.ServeTLS() errors back to the caller

### PR DESCRIPTION
**Pull request by alessio**
When http.ServeTLS() fails, StartHTTPAndTLSServer() logs the error then hangs for ever.
It should instead pass the error back to the caller.

Closes: #2460

* [x] Updated all relevant documentation in docs
* [x] Updated all code comments where relevant
* [x] Wrote tests
* [x] Updated CHANGELOG_PENDING.md

### CODE REVIEW

File: rpc/lib/server/http_server.go [hunk 1]
```diff
@@ -102,15 +102,16 @@ func StartHTTPAndTLSServer(
 		listener = netutil.LimitListener(listener, config.MaxOpenConnections)
 	}
 
-	go func() {
```
Comment (melekes) [inline rpc/lib/server/http_server.go]: I will ask you to apply the same changes to `StartHTTPServer` func. We need to keep them consistent, so we can swap HTTP server with HTTPS.

Comment (alessio) [reply rpc/lib/server/http_server.go]: Will do shortly

Comment (alessio) [reply rpc/lib/server/http_server.go]: @melekes Done

Comment (alessio) [reply rpc/lib/server/http_server.go]: mmm tests failed, I'm on it

**Pull request merged**